ds3-autogen-go/src/main/resources/tmpls/go/request/with_headers.ftl

```diff
@@ -0,0 +1,12 @@
+func (${name?uncap_first} *${name}) WithMetaData(key string, value string) *${name} {
+    if strings.HasPrefix(strings.ToLower(key), AMZ_META_HEADER) {
+        ${name?uncap_first}.headers.Add(key, value)
```

Comment:
So lower case them in our code, and let the networking library canonicalize them?